WeakAuras/Prototypes.lua

```diff
@@ -6904,19 +6903,67 @@ Private.event_prototypes = {
                     end
                   end
                 end
-              ]]
-              ret = ret .. ret2:format(index, value and "true" or "false")
+              ]]):format(index, value and "true" or "false"))
             end
           end
           if (inverse) then
-            ret = ret .. [[
-            active = not (active);
-            ]]
+            table.insert(ret, [[
+              active = not (active);
+            ]])
           end
-          return ret;
         end
       end
-      return "";
+      if (WeakAuras.IsTWW() and trigger.use_herotalent == false) then
+        if (trigger.talent.multi) then
+          table.insert(ret, [[
+            local heroActive = false;
+            local activeIcon, activeName, _
+            local index
+            local firstLoop = true
+          ]])
+          for index, value in pairs(trigger.herotalent.multi) do
+            table.insert(ret, ([[
+              local talentId = %s
+              local shouldBeActive = %s
+              if talentId then
+                activeName, activeIcon, _, rank = WeakAuras.GetTalentById(talentId)
+                if activeName ~= nil then
+                  local hasTalent = rank > 0
+                  if hasTalent then
```

Comment:
I think this can all be simplified to:

Before the loop:
local heroActive = true

if (rank > 0) ~= shouldBeActive then
  heroActive = false
  break
end

Now, that changes behaviour if no herotalent is selected at all, if you want to preserve that behaviour, the easiest is imho to simply do that at the very top level, thus e.g.:

if (next(trigger.herotalent.multi) == nil) then
  table.insert(ret, [[
    local heroActive = false
 ]])
else 
  [[ loop over talents ]]
end

Reply:
done